dagshub/data_engine/client/gql_introspections.py

```diff
@@ -7,55 +7,70 @@
 class GqlIntrospections:
     @staticmethod
     @functools.lru_cache()
-    def input_fields() -> str:
+    def obj_fields() -> str:
         q = (
             GqlQuery()
             .operation("query", name="introspection")
-            .query("__schema").fields([
-                GqlQuery().fields(name="types", fields=[
-                    "name",
-                    GqlQuery().fields(name="inputFields", fields=[
-                        "name",
-                        GqlQuery().fields(name="type", fields=[
-                            "name"
-                        ]).generate()
-                    ]).generate()
-                ]).generate()
-            ]).generate()
+            .query("__schema")
+            .fields(
+                [
+                    GqlQuery()
+                    .fields(
+                        name="types",
+                        fields=[
+                            "name",
+                            GqlQuery()
+                            .fields(
+                                name="inputFields",
+                                fields=["name", GqlQuery().fields(name="type", fields=["name"]).generate()],
+                            )
+                            .generate(),
+                            GqlQuery()
+                            .fields(
+                                name="fields",
+                                fields=["name", GqlQuery().fields(name="type", fields=["name"]).generate()],
+                            )
+                            .generate(),
+                        ],
+                    )
+                    .generate()
+                ]
+            )
+            .generate()
         )
         return q
 
 
 @dataclass
-class InputField:
+class Field:
     name: str
 
 
 @dataclass
 class IntrospectionType:
     name: str
-    inputFields: List[InputField]
+    inputFields: List[Field]
+    fields: List[Field]
+
+    def __hash__(self):
+        return hash(self.hash_str())
+
+    def hash_str(self):
+        return self.name + "".join([f.name for f in self.fields]) + "".join([f.name for f in self.inputFields])
```

Comment:
If you're doing all the hash stuff for the contain check at line 104, then it won't really work, you need to put them in a set and not in a list for that. List containment doesn't utilize hashing iirc.

Reply:
Unless I'm missing something and there's another reason you're doing the hashing. Please point me at that if it is so

Reply:
It’s necessary for functools lru_cache to work since it’s passed as an argument to a cached function.